crates/solidity/outputs/cargo/crate/src/lib.rs

```diff
@@ -3,44 +3,68 @@ mod generated;
 pub use generated::*;
 
 #[cfg(feature = "__experimental_bindings_api")]
-pub fn transform_built_ins_node(node: &generated::cst::Node) -> generated::cst::Node {
-    use std::rc::Rc;
-
-    use generated::cst::{Edge, Node, NonterminalNode, TerminalNode};
-
-    use crate::cst::TerminalKind;
-
-    match node {
-        Node::Nonterminal(nonterminal) => {
-            let NonterminalNode {
-                kind,
-                text_len,
-                children,
-            } = nonterminal.as_ref();
-            let children = children
-                .iter()
-                .map(|edge| Edge {
-                    label: edge.label,
-                    node: transform_built_ins_node(&edge.node),
-                })
-                .collect();
-            let nonterminal = Rc::new(NonterminalNode {
-                kind: *kind,
-                text_len: *text_len,
-                children,
-            });
-            Node::Nonterminal(nonterminal)
-        }
-        Node::Terminal(terminal) => {
-            let TerminalNode { kind, text } = terminal.as_ref();
-            let terminal = match terminal.as_ref().kind {
-                TerminalKind::Identifier => Rc::new(TerminalNode {
+pub mod bindings {
```

Comment:
I was thinking about how we can have language-specific extensions like this, while still make it usable from the generic runtime. I had to create a similar mechanism for the `Compilation` API.

WDYT of having an `extensions` sub-module, that is not copied over as part of the runtime? For example:

- add a `crates/codegen/runtime/cargo/crate/src/extensions/bindings/mod.rs` that defines an empty `fn add_built_ins()`. The whole `extensions` dir won't be copied over.
- In `crates/codegen/runtime/cargo/crate/src/lib.rs` just add `mod extensions;`.
- In `create_with_resolver()` call `crate::extensions::bindings::add_built_ins()` directly.
- Add a real `add_built_ins()` in `crates/solidity/outputs/cargo/crate/src/extensions/bindings/mod.rs` that will do the solidity specific stuff.

This way, it stays hidden/internal, and users like tests (or WASM in my case) don't have to worry about invoking it. It also stays consistent between all languages.